### PR TITLE
replace apex.normalization.FusedLayerNorm with torch.nn.LayerNorm

### LINKED ISSUE #9377: replacing apex.normalization.FusedLayerNorm with torch.nn.LayerNorm
It seems that time has arrived to drop `apex.normalization.FusedLayerNorm` in favor of `torch.nn.LayerNorm` 

1. the latter was ported more than a year ago from apex  https://github.com/pytorch/pytorch/pull/27634 (around pt-1.4)
2. it's faster than the apex according to my benchmarks https://github.com/pytorch/pytorch/issues/37713#issuecomment-753434842 (**33% faster on rtx-3090!**, 10% faster on gtx-1070)

**but note:** this same benchmark run here https://github.com/pytorch/fairseq/issues/2012#issuecomment-622607286 on V100 reports the opposite  - that the native is slower (pt-1.5). So it might help to run this very quick benchmark on other cards and compare. In particular if you have access to V100 please report back the findings at this thread: https://github.com/pytorch/pytorch/issues/37713

The main reason for this need is that `apex.normalization.FusedLayerNorm` is buggy (corrupts memory) when it comes to switching devices, which is done a lot under Model Parallel. https://github.com/NVIDIA/apex/issues/1022
With  `apex.normalization.FusedLayerNorm` things fail a lot under MP and requires sticking `torch.cuda.set_device(id)` in many many places as a workaround :( Since this overload is used at model's init time it's not possible to not use it under MP as the latter gets activate after model's init.

I will use that workaround if you find out that apex is faster still on some important-to-consider hardware. And, of course, in that case please report back to the pytorch team so that they could fix it. Otherwise apex support is pretty much no more and it's just a matter of time before apex will be unusable.

The models that need that change are bart/fsmt/prophetnet.

@patrickvonplaten, @LysandreJik

### PR DESCRIPTION
**Pull request by stas00**
This PR proposes to drop `apex.normalization.FusedLayerNorm` in favor of faster `torch.nn.LayerNorm`. 

1. For performance and background details please see the discussions in https://github.com/huggingface/transformers/issues/9377
2. It's also needed for https://github.com/huggingface/transformers/pull/9384 since `apex.normalization.FusedLayerNorm` corrupts data under model parallel https://github.com/NVIDIA/apex/issues/1022

Fixes: #9377

@LysandreJik, @sgugger, @patrickvonplaten

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]
This looks good to me!

**Review by LysandreJik — APPROVED** [review]
LGTM too!

**Review by sgugger — APPROVED** [review]
Way easier this way, LGTM!

**Comment by patrickvonplaten** [comment]
Merging since it's blocking #9343 .
